apps/dotcom/src/components/ShareMenu.tsx

```diff
@@ -124,20 +124,22 @@ export const ShareMenu = React.memo(function ShareMenu() {
 			})
 		}
 
-		getReadonlyUrl().then((readonlyUrl) => {
-			if (readonlyUrl && !shareState.readonlyQrCodeDataUrl) {
-				// fetch the readonly QR code data URL
-				createQRCodeImageDataString(readonlyUrl).then((dataUrl) => {
-					if (!cancelled) {
-						setShareState((s) => ({ ...s, readonlyUrl, readonlyQrCodeDataUrl: dataUrl }))
-					}
-				})
-			}
-		})
+		if (!shareState.readonlyUrl) {
+			getReadonlyUrl().then((readonlyUrl) => {
+				if (readonlyUrl) {
+					// fetch the readonly QR code data URL
+					createQRCodeImageDataString(readonlyUrl).then((dataUrl) => {
+						if (!cancelled) {
+							setShareState((s) => ({ ...s, readonlyUrl, readonlyQrCodeDataUrl: dataUrl }))
+						}
+					})
+				}
+			})
+		}
 
 		const interval = setInterval(() => {
 			const url = window.location.href
-			if (shareState.url === url) return
+			if (decodeURIComponent(url) === decodeURIComponent(shareState.url)) return
```

Comment:
digging a little more into this - looks like we should fix more of an underlying problem here.

in sharing.ts, line 141: we should do:
```navigate(decodeURIComponent(pathname))```

the problem is that `new URLSearchParams` a couple lines before that encodes things unnecessarily (for us) and then when we navigate it's actually doing two navigations, to, for example:
1.) http://localhost:3000/r/3uc5l9JkT2dob3Q8DFr43?v=0%2C0%2C1460%2C679&p=page
2.) http://localhost:3000/r/3uc5l9JkT2dob3Q8DFr43?v=0,0,1460,679&p=page

you can see this if you put some breakpoints in as it navigates.

so anyway, instead of adding `decodeURIComponent` let's fix up sharing.ts!

Reply:
What if we do both? I can see us making the same kind of mistake again and it could take us a while to discover. Keeping the decoding where we do the check could prevent such future problems.

Reply:
My hesitation is that it really should be `decodeURI` (since it's looking at the whole URI and not just the pathname).
And then that actuall ywouldn't have fixed this issue :-/

Reply:
Added a helper function to  ccompare urls. Might come handy in the future. Can also drop the whole check, but it still feels worth it to me.